src/pfe/file-watcher/server/test/functional-test/functional.test.ts

```diff
@@ -33,64 +43,57 @@ describe("Functional Test Suite", () => {
     });
   });
 
-  for (const chosenType of projectTypes) {
-    if (chosenType === "docker") {
-      for (const chosenDocker of dockerProjects) {
-        cloneProject(chosenType, chosenDocker);
-      }
-    } else {
-      cloneProject(chosenType);
-    }
-  }
-
-  before("set up test resources", () => {
-    utils.createFWDataDir();
+  before("registering socket listener", () => {
+    socket.registerListener("PFE Functional Test Listener");
   });
 
-  after("clean up test resources", () => {
-    utils.cleanWorkspace();
-    utils.removeFWDataDir();
-    // remove logs dir only for now till the delete project test has been added
-    utils.cleanUpLogsDir();
-  });
+  if (process.env.IN_K8) {
+    before("set deployment registry", function (done: any): void {
+      this.timeout(timoutConfigs.defaultTimeout);
+      const workspace_settings_file_content = { deploymentRegistry: pfe_configs.deploymentRegistry };
+      const workspace_settings_file_content_json = JSON.stringify(workspace_settings_file_content);
+      const workspace_settings_file = path.join(app_configs.codewindWorkspaceDir, ".config", "settings.json");
 
-  describe("generic suite", () => {
-    describe("setLocale", () => {
-      localeTestSuite();
+      utils.writeToFile(workspace_settings_file, workspace_settings_file_content_json, (err) => {
+        if (err) done(err);
+        fs.chown(workspace_settings_file, 1001, 2004, (err) => {
+          if (err) done(err);
+          utils.getRegistry((err, res, body) => {
+            if (err) done(err);
+            expect(res);
```

Comment:
If you want to check the target(variable/object) existence, Mocha has `expect(res).to.exist;` for checking both `null` and `undefined`. It's a general comment and can apply to all lines for checking the target(variable/object) existence

Reply:
`expect(res);` and `expect(res).to.exist;` are the exact same.

Reply:
I am not sure if `expect(res);` is proper way to use, but from the Chai man page: https://www.chaijs.com/api/bdd/, `expect(res).to.exist;` is the official way to check existence and it's more readable

Reply:
Yes, logically they do the exact same. If you want I can change it but they do the exact same and it's in a lot of places so will be tedious.

Reply:
I see, I think we can do this improvement when we extend the test cases for other project types. It's fine for this PR.